models/tshirt/include/Tshirt.h

```diff
@@ -0,0 +1,289 @@
+        const double depth = 2.0;         //!< Total soil column depth ('D') [m]
+
+        //! Constructor for tshirt parameters
+        /*!
+            Constructor for tshirt param objects.
+        */
+        tshirt_params(double maxsmc, double wltsmc, double satdk, double satpsi, double slope, double b,
+                double multiplier, double alpha_fc, double Klf, double Kn, int nash_n, double Cgw, double expon) :
+                maxsmc(maxsmc),
+                wltsmc(wltsmc),
+                satdk(satdk),
+                satpsi(satpsi),
+                slope(slope),
+                b(b),
+                multiplier(multiplier),
+                alpha_fc(alpha_fc),
+                Klf(Klf),
+                Kn(Kn),
+                nash_n(nash_n),
+                Cgw(Cgw),
+                expon(expon)
+        {
+            this->Ssmax = this->depth * this->maxsmc;
+        }
+
+    };
+
+    /*!
+     * Tshirt state structure
+     *
+     * This structure provides storage for the state used by Tshirt hydrological model at a particular time step.
+     */
+    struct tshirt_state
+    {
+        // TODO: confirm this is correct
+        double Ss;           //!< current water storage in soil column nonlinear reservoir
+        double Sgw;          //!< current water storage in ground water nonlinear reservoir
+        double* Snash;       //!< water storage in nonlinear reservoirs of Nash Cascade for lateral subsurface flow
+
+        // I think this doesn't belong in state, and so is just in run() below
+        //double column_total_soil_moisture_deficit;    //!< soil column total moisture deficit
+
+        tshirt_state(double ss, double sgw, double* nash_res_ptr = 0x0) : Ss(ss), Sgw(sgw), Snash(nash_res_ptr) {}
+    };
+
+    /*!
+     * Tshirt flux structure
+     *
+     * This structure provides storage for the fluxes generated by Tshirt at any time step
+     */
+    struct tshirt_fluxes
+    {
+        double surface_runoff;  //!< Direct surface runoff
+        double Qgw;             //!< Groundwater flow from groundwater reservoir to channel flow
+        double Qperc;           //!< Percolation flow from subsurface to groundwater reservoir
+        double Qlf;             //!< Lateral subsurface flow
+        double et_loss;         //!< Loss from ET
+
+        tshirt_fluxes(double q_gw, double q_perc, double q_lf, double runoff, double et_loss) : Qgw(q_gw),
+                                                                                                Qperc(q_perc),
+                                                                                                Qlf(q_lf),
+                                                                                                surface_runoff(runoff),
+                                                                                                et_loss(et_loss) {
+
+        }
+    };
+
+    // TODO: consider combining with or differentiating from similar hymod enum
```

Comment:
common error represntations, i.e. mass_balance_error, can be reused and probably should be.  Unique conditions per model can can be enumerated in the model namespace

Reply:
This can be done later